caravel/assets/visualizations/filter_box.jsx

```diff
@@ -10,22 +10,80 @@ import '../stylesheets/react-select/select.less';
 
 import './filter_box.css';
 
+const TIME_CHOICES = [
+  '1 hour ago',
+  '12 hours ago',
+  '1 day ago',
+  '7 days ago',
+  '28 days ago',
+  '90 days ago',
+  '1 year ago',
+];
+
+const propTypes = {
+  origSelectedValues: React.PropTypes.object,
+  filtersChoices: React.PropTypes.object,
+  onChange: React.PropTypes.func,
+  showDateFilter: React.PropTypes.bool,
+};
+
+const defaultProps = {
+  origSelectedValues: {},
+  onChange: () => {},
+  showDateFilter: false,
+};
+
 class FilterBox extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       selectedValues: props.origSelectedValues,
     };
   }
+  changeFilter(filter, options) {
+    let vals = null;
+    if (options) {
+      if (Array.isArray(options)) {
```

Comment:
because we use a mix of Select multiple and Select one... The time ones are single choice